ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/binary_ng_program_factory.cpp

```diff
@@ -121,8 +121,12 @@ void set_or_update_runtime_arguments(
             std::array compute_runtime_args = {num_tiles_per_core, freq, counter};
             handle_args(program, compute_kernel_id, core, compute_runtime_args);
         } else {
-            class bfloat16 bfloat_scalar(*operation_attributes.scalar);
-            uint32_t packed_scalar = pack_two_bfloat16_into_uint32({bfloat_scalar, bfloat_scalar});
+            const auto scalar = *operation_attributes.scalar;
+            class bfloat16 bfloat_scalar(scalar);
+            const auto packed_scalar = a.get_dtype() == DataType::FLOAT32 ? std::bit_cast<uint32_t>(scalar)
+                                       : a.get_dtype() == DataType::INT32
+                                           ? std::bit_cast<uint32_t>(static_cast<int32_t>(scalar))
+                                           : pack_two_bfloat16_into_uint32({bfloat_scalar, bfloat_scalar});
```

Comment:
`bfloat16` is implicitly constructible from `float`, so this should be sufficient:
```suggestion
            const auto scalar = *operation_attributes.scalar;
            const auto packed_scalar = a.get_dtype() == DataType::FLOAT32 ? std::bit_cast<uint32_t>(scalar)
                                       : a.get_dtype() == DataType::INT32
                                           ? std::bit_cast<uint32_t>(static_cast<int32_t>(scalar))
                                           : pack_two_bfloat16_into_uint32({scalar, scalar});
```